anitya/lib/backends/github.py

```diff
@@ -59,18 +65,168 @@ def get_versions(cls, project):
             when the versions cannot be retrieved correctly
 
         '''
+        owner = None
+        repo = None
         if project.version_url:
-            url_template = 'https://github.com/%(version_url)s/tags'
-            version_url = project.version_url.replace(
-                'https://github.com/', '')
-            url = url_template % {'version_url': version_url}
+            url = project.version_url
         elif project.homepage.startswith('https://github.com'):
-            url = project.homepage
+            url = project.homepage.replace('https://github.com/', '')
             if url.endswith('/'):
-                url = project.homepage[:-1]
-            url += '/tags'
+                url = url[:-1]
         else:
             raise AnityaPluginException(
                 'Project %s was incorrectly set-up' % project.name)
 
-        return get_versions_by_regex(url, REGEX, project)
+        if url:
+            (owner, repo) = url.split('/')
+
+        if (not owner) or (not repo):
+            raise AnityaPluginException(
+                'Project %s was incorrectly set-up. \
+                Can\'t parse owner and repo' % project.name)
+
+        query = cls.prepare_query(owner, repo)
+
+        try:
+            resp = cls.call_url_post(API_URL, query, use_token=True)
+        except Exception as err:
+            _log.debug('%s ERROR: %s' % (project.name, str(err)))
+            raise AnityaPluginException(
+                'Could not call : "%s" of "%s", with error: %s' % (
+                    url, project.name, str(err)))
+
+        if resp.ok:
+            json = resp.json()
+        else:
+            raise AnityaPluginException(
+                '%s: Server responded with status "%s": "%s"' % (
+                    project.name, resp.status_code, resp.reason))
+
+        if 'errors' in json:
```

Comment:
Consider making a function that accepts the json and spits out the versions (or raises an exception). That'll make testing easier, and make this function a digestible length.

Reply:
Ok, I will do this. It sounds good.